src/extension.ts

```diff
@@ -0,0 +1,62 @@
+import * as vscode from 'vscode';
+
+import { startServer } from './lib/server';
+import {
+  handleHubspotConfigPostRequest,
+  registerConfigDependentFeatures,
+} from './lib/auth';
+import { registerCommands } from './lib/commands';
+import { initializeStatusBar } from './lib/statusBar';
+import { getDefaultPortalFromConfig, getRootPath } from './lib/helpers';
+import {
+  getUpdateLintingOnConfigChange,
+  setLintingEnabledState,
+} from './lib/lint';
+import { PortalsProvider } from './lib/providers/portalsProvider';
+import { HubspotConfig } from './lib/types';
+import { COMMANDS, TREE_DATA } from './lib/constants';
+
+export const activate = async (context: vscode.ExtensionContext) => {
+  console.log('Activating Extension...');
+  const rootPath = getRootPath();
+
+  registerCommands(context);
+  initializeStatusBar(context);
+  registerConfigDependentFeatures(
+    context,
+    rootPath,
+    (configPath: string) => {
+      console.log('loadConfigDependentFeatures');
+      setLintingEnabledState();
+      context.subscriptions.push(getUpdateLintingOnConfigChange());
+      const portalProvider = new PortalsProvider();
+      context.subscriptions.push(
+        vscode.commands.registerCommand(COMMANDS.PORTALS_REFRESH, () => {
+          console.log(COMMANDS.PORTALS_REFRESH);
+          portalProvider.refresh();
+        })
+      );
+      vscode.window.registerTreeDataProvider(TREE_DATA.PORTALS, portalProvider);
+    },
+    (config: HubspotConfig, configPath: string) => {
+      console.log(
+        'updateConfigDependentFeatures',
+        config.defaultPortal,
+        JSON.stringify(config.portals.map((p) => p.name)),
+        configPath
+      );
+      vscode.commands.executeCommand(
+        COMMANDS.CONFIG_SET_DEFAULT_ACCOUNT,
+        getDefaultPortalFromConfig(config),
+        { silenceNotification: true }
+      );
+      vscode.commands.executeCommand(COMMANDS.PORTALS_REFRESH);
+    }
+  );
+
+  startServer({
+    onPostRequest: async (req: any) => {
+      return handleHubspotConfigPostRequest(req, { rootPath });
+    },
+  });
```

Comment:
Is this always running? I think it would be best to only start this server up when needed